nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -181,23 +182,83 @@ public IcebergTable table(TableIdentifier tableIdentifier) {
   }
 
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    getRef().checkMutable();
+
+    if (namespace.isEmpty()) {
+      throw new IllegalArgumentException("Creating empty namespaces is not supported");
+    }
+
+    ContentKey key = ContentKey.of(namespace.levels());
+    org.projectnessie.model.Namespace content =
+        org.projectnessie.model.Namespace.of(key.getElements(), metadata);
+
     try {
-      getRef().checkMutable();
-      withReference(
-              getApi()
-                  .createNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
-                  .properties(metadata))
-          .create();
-      refresh();
-    } catch (NessieNamespaceAlreadyExistsException e) {
-      throw new AlreadyExistsException(e, "Namespace already exists: %s", namespace);
+
+      // First, check that the namespace doesn't exist at the current hash;
+      // this will avoid a NessieBadRequestException when committing the namespace creation
+      Reference ref = getReference();
+      Map<ContentKey, Content> contentMap = api.getContent().reference(ref).key(key).get();
+      Content existing = contentMap.get(key);
+      if (existing != null) {
+        throwNamespaceAlreadyExists(key, existing);
+      }
+
+      try {
+
+        CommitMultipleOperationsBuilder commitBuilder =
+            api.commitMultipleOperations()
+                .commitMeta(
+                    NessieUtil.buildCommitMetadata("create namespace " + key, catalogOptions))
+                .operation(Operation.Put.of(key, content));
+
+        Tasks.foreach(commitBuilder)
+            .retry(5)
+            .stopRetryOn(
+                NessieReferenceNotFoundException.class, NessieReferenceConflictException.class)
+            .throwFailureWhenFinished()
+            .onFailure((o, exception) -> refresh())
+            .run(
+                b -> {
```

Comment:
Nit: rename `b` to `commit` or so (`b` can be associated with a branch)

Reply:
I'll use `builder` if you don't mind, `commit` looks a bit weird too : `commit.branch(...).commit()`